cmd/export-proxy/main.go

```diff
@@ -0,0 +1,187 @@
+
+			config, err := config.Active()
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to load active configuration")
+			} else if config.Registry.Token == "" {
+				log.FromContext(ctx).Fatal("Active configuration doesn't have a GCP access token")
+			}
+
+			client, err := connection.NewRegistryClient(ctx)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get client")
+			}
+
+			name, err := names.ParseDeployment(args[0])
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to parse deployment")
+			}
+
+			deployment, err := client.GetApiDeployment(ctx, &rpc.GetApiDeploymentRequest{
+				Name: name.String(),
+			})
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get deployment")
+			}
+
+			root := APIProxy{
+				Name:           name.DeploymentID,
+				DisplayName:    deployment.DisplayName,
+				Description:    deployment.Description,
+				BasePath:       "/",
+				ProxyEndpoint:  "default",
+				TargetEndpoint: "default",
+			}
+
+			proxy := ProxyEndpoint{
+				Name:     "default",
+				BasePath: "/",
+				RouteRule: RouteRule{
+					Name:           "default",
+					TargetEndpoint: "default",
+				},
+			}
+
+			target := TargetEndpoint{
+				Name: "default",
+				URL:  deployment.EndpointUri,
+			}
+
+			proxyZip, err := bundle(root, proxy, target)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to create bundle")
+			}
+
+			if err := createProxy(ctx, root.Name, proxyZip.Bytes(), config.Registry.Token); err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to create proxy")
+			}
+		},
+	}
+
+	ctx := context.Background()
+	if err := cmd.ExecuteContext(ctx); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func bundle(root APIProxy, proxy ProxyEndpoint, target TargetEndpoint) (bytes.Buffer, error) {
+	var buf bytes.Buffer
+	zw := zip.NewWriter(&buf)
+
+	if err := write(zw, "apiproxy/"+root.Name+".xml", root); err != nil {
+		return buf, err
+	}
+
+	if err := write(zw, "apiproxy/proxies/default.xml", proxy); err != nil {
+		return buf, err
+	}
+
+	if err := write(zw, "apiproxy/targets/default.xml", target); err != nil {
+		return buf, err
+	}
+
+	if err := zw.Close(); err != nil {
+		return buf, err
+	}
+
+	return buf, nil
+}
+
+func write(zw *zip.Writer, name string, v any) error {
+	out, err := xml.MarshalIndent(v, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	w, err := zw.Create(name)
+	if err != nil {
+		return err
+	}
+
+	if _, err := w.Write([]byte(xml.Header)); err != nil {
+		return err
+	}
+
+	if _, err := w.Write(out); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createProxy(ctx context.Context, name string, bundle []byte, token string) error {
+	req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("https://apigee.googleapis.com/v1/organizations/egansean-integrations1/apis?action=import&name=%s", name), bytes.NewReader(bundle))
```

Comment:
Minor: we'll need a way to customize this org name.

Reply:
Oh I didn't mean to leave that hardcoded in. Will fix.